action_server/src/action_server/client.py

```diff
@@ -76,12 +77,18 @@ def send_task(self, semantics):
         recipe = semantics
 
         result = None
-        while not result:
-            goal = action_server.msg.TaskGoal(recipe=recipe)
-            self._action_client.send_goal(goal)
+        goal = action_server.msg.TaskGoal(recipe=recipe)
+        self._action_client.send_goal(goal)
+
+        try:
             self._action_client.wait_for_result()
             result = self._action_client.get_result()
 
+        # if user presses ctrl+C, stop waiting and cancel all goals
+        except KeyboardInterrupt:
+            self.cancel_all()
+            return TaskOutcome(TaskOutcome.RESULT_CANCELED)
```

Comment:
I think that we should not return a TaskOutcome with a specific result "canceled", but the Keyboardinterrupt should be raised after canceling the goals. This leaves it up to the user of the client class to determine whether the KeyboardInterrupt should end the managing program or not. Example: In the Natural Language Console, we may not want a Ctrl+C to stop the program, but during a challenge, a single Ctrl+C in the challenge-terminal should stop the entire challenge, not just the current task.